src/shipit_static_analysis/shipit_static_analysis/revisions.py

```diff
@@ -101,15 +118,21 @@ def build_diff_name(self):
     def url(self):
         return 'https://{}/{}/'.format(self.api.hostname, self.diff_phid)
 
-    def apply(self, repo):
+    def load(self, repo):
```

Comment:
In the Phabricator case, you are assuming the Phabricator patch applies to the current mozilla-central, right?

Reply:
This could be a problem, as there are three cases where this might not work:
1) A Phabricator patch applies on top of a mozilla-central revision that isn't `tip`; (I assume this is very common)
2) A Phabricator patch applies on top of other Phabricator patches; (this can be common too)
1) A Phabricator patch applies on top of a non-published revision. (this should be pretty rare and there's no way we can handle it, so we should ignore it).

Reply:
Yes, but this is already the case, as we must use the API to retrieve the patch (supporting multiple checkout hg/git methods would be too much work and prone to bugs).

Here the workflow related to Phabricator did not change, i try to merge on top of current tip, which could indeed not work all the time.
Another solution would be to merge on the ancestor of the commit, but i would need to get it from mercurial (easy), and from git (not so easy, even though i saw an API doing hg<->git revision translations for MC).

Finally the phabricator publication is not used right now, and would need some polishing...